app/src/lib/stores/helpers/branch-pruner.ts

```diff
@@ -190,8 +193,24 @@ export class BranchPruner {
       mb => !ReservedRefs.includes(mb.canonicalRef)
     )
 
+    // get the local remote ref name for the github repository attached to this repo
+    const defaultRemoteRefName = (await getRemotes(this.repository)).find(
+      r =>
+        r.url === gitHubRepository.cloneURL ||
+        r.url === gitHubRepository.cloneURL
+    )
+
+    // get the locally cached branches of remotes (ie `remotes/origin/master`)
+    const remoteBranches = (await getBranches(
+      this.repository,
+      `refs/remotes/${defaultRemoteRefName}`
+    )).map(b => formatAsLocalRef(b.name))
+
     const branchesReadyForPruning = candidateBranches.filter(
-      mb => !recentlyCheckedOutCanonicalRefs.has(mb.canonicalRef)
+      candidate =>
+        !recentlyCheckedOutCanonicalRefs.has(candidate.canonicalRef) &&
+        !remoteBranches.includes(candidate.canonicalRef)
+      // do i need to have the upstream of the merged branch to make this check?
```

Comment:
I'd prefer this check looked at the `upstream` of the branch because the names don't have to match when tracking an upstream branch in Git.

Reply:
yeah, you are totally right. good shout.